File: perl/get_version.pl
#!/usr/bin/perl

use strict;

my $topdir = $ARGV[0];

open IN,"<$topdir/configure.ac" or die("Can't open \"$topdir/configure.ac\": $!");

my $line;
my $version;
while($line = <IN>) {
	chomp($line);
	if ($line =~ m/AC_INIT\(coan,\W*(.+)\)/) {
		$version = $1;
		last;
	}
}
close IN;
if (defined($version)) {
	print $version;
	exit(0);
}
print STDERR "Failed to parse coan version from \"$topdir/configure.ac\"\n";
exit(1);

File: python/get_version.py
#!/usr/bin/python

import sys, re
from coanlib import fopen

topdir = sys.argv[1];
version = None

fh = fopen(topdir + '/configure.ac','r')
for line in fh:
	text = line.rstrip();
	match = re.search('AC_INIT\(coan,\W*(.+)\)',text)
	version = match.group(1)
	if version:
		break
		
fh.close()

if version:
	print version
	sys.exit(0);

sys.stderr.write('Failed to parse coan version from \"' + topdir \
	+ '/configure.ac\"\n')
sys.exit(1);

